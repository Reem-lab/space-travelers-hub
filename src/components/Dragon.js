import React from 'react';
import propTypes from 'prop-types';

import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Button from 'react-bootstrap/Button';
import Card from 'react-bootstrap/Card';
import 'bootstrap/dist/css/bootstrap.min.css';

import { MDBContainer } from 'mdbreact';

const Dragon = (props) => {
  const { dragon } = props;

  return (

    <Row id={dragon.id} className="px-4">
      <Col sm={3} className="d-flex align-items-center p-0">
        <Card.Img src={dragon.flickr_images[0]} />
      </Col>
      <Col sm={9} className="d-flex flex-column align-items-start">
        <Card.Title>
          {dragon.name}
          <span className="mx-4 h6">
            Type:
            {dragon.type}
          </span>
        </Card.Title>
        <MDBContainer className="p-0 flex-1">
          <Card.Text className="text-left">
            <span className="mx-4 h6">Reserved</span>
            {dragon.description}
          </Card.Text>
        </MDBContainer>
        <Button>Reserve Dragon</Button>
      </Col>
      <hr />
    </Row>
  );
};

Dragon.propTypes = {
  dragon: propTypes.objectOf(String).isRequired,
};

export default Dragon;
