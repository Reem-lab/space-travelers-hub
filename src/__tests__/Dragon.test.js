import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Dragon from '../components/Dragon';

// ARRANGE
const dragons = [
  {
    id: 'dragon 5',
    name: 'dragonite',
    type: 'big capsule',
    description: 'ready to go to space',
    reserved: false,
  },
];

describe('Test render the Dragon Component', () => {
  it('Dragons section renders a hardcoded object, passed as a prop', () => {
    // ACT
    const { container } = render(
      <Provider store={store}>
        <Dragon key={dragons[0].id} dragon={dragons[0]} />
      </Provider>,
    );

    // ASSERT
    expect(container).toMatchSnapshot();
  });
});
