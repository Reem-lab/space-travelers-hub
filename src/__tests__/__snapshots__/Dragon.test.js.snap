// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render the Dragon Component Dragons section renders a hardcoded object, passed as a prop 1`] = `
<div>
  <div
    class="px-4 row"
    id="dragon 5"
  >
    <div
      class="d-flex align-items-center p-0 col-sm-3"
    >
      <img
        class="card-img"
      />
    </div>
    <div
      class="d-flex flex-column align-items-start col-sm-9"
    >
      <div
        class="card-title h5"
      >
        dragonite
        <span
          class="mx-4 h6"
        >
          Type:
          big capsule
        </span>
      </div>
      <div
        class="container p-0 flex-1"
        data-test="container"
      >
        <p
          class="text-left card-text"
        >
          <span
            class="mx-4 h6 
            d-none"
          >
            Reserved
          </span>
          ready to go to space
        </p>
      </div>
      <button
        class="btn btn-primary"
        type="button"
      >
        Reserve Dragon
      </button>
    </div>
    <hr />
  </div>
</div>
`;
